Editor bugs: Grid is missing, incorrect line numbers in scripts

Version: https://github.com/utopia-rise/godot-kotlin-jvm/releases/tag/0.1.4-3.2.3
Platform: Ubuntu 20.04.2 LTS
Editor: godot-kotlin-jvm_editor_x11_.zip

First, I want to say that your project is amazing. I really love Kotlin and the ability to use this language in Godot makes me very happy!

I am trying to work with your build of godot editor and encountering several problems: 

1. Editor grid is missing:
![image](https://user-images.githubusercontent.com/3995047/115548995-4f2b4600-a2d2-11eb-82f5-a6b709bc122d.png)

2. Incorrect line numbers in all scripts (include GDScript):
![image](https://user-images.githubusercontent.com/3995047/115549055-666a3380-a2d2-11eb-9ae3-2517fb336c2c.png)

3. Remote tree of running game also doesn't work. Maybe this log will help:

`
Running: /home/sipaha/Programs/godot/godot.x11.opt.tools.64 --path /home/sipaha/GodotProject --remote-debug 127.0.0.1:6007,000000 --allow_focus_steal_pid 157961 --position 448,240 res://Spatial.tscn
ERROR: to_int: Cannot represent 6007,000000 as integer, provided value is too big.
   At: core/ustring.cpp:1794.
ERROR: connect_to_host: Remote Debugger: Unable to connect. Status: 3.
   At: scene/debugger/script_debugger_remote.cpp:95.
`

I am checked this builds:
1. last release (godot-kotlin-jvm/releases/tag/0.1.4-3.2.3) - problems exist
2. build from sources with kotlin module - problems exists
3. build from same engine sources, but without kotlin module - problems doesn't exists
4. last official release - problems doesn't exists

> Ok got the same on apple silicon.
> If you run in verbose mode from terminal, do you have this printed ?
> 
> ```
> ERROR: _display_error_with_code: SceneShaderGLES3: Vertex Program Compilation Failed:
> ERROR: 0:470: Too many arguments to constructor of 'vec3'
> ERROR: 0:470: Too many arguments to constructor of 'vec3'
> ERROR: 0:470: Too many arguments to constructor of 'vec3'
> ```
@piiertho I have very similar error messages:
```
Godot-JVM: Godot-JVM: Editor mode, loading jvm from JAVA_HOME
Godot-JVM: Starting JVM ...
Godot-JVM: Loading bootstrap jar: /home/sipaha/Programs/godot/godot-bootstrap.jar
WARNING: JNI local refs: 33, exceeds capacity: 32
Godot-JVM: Starting GC thread
Godot-JVM: WARNING: No main.jar detected. No classes will be loaded. Build the gradle project to load classes
______SHADER_OUTPUT_____ 
ERROR: _display_error_with_code: SceneShaderGLES3: Vertex Program Compilation Failed:
0(470) : error C1068: too much data in type constructor
0(470) : error C1068: too much data in type constructor
0(470) : error C1068: too much data in type constructor
```
[editor-floats-error-log.txt](https://github.com/utopia-rise/godot-kotlin-jvm/files/6362006/editor-floats-error-log.txt)

--- a/.gitignore
+++ b/.gitignore
@@ -7,3 +7,5 @@
 *.o
 __pycache__
 libprotobuf/
+
+**/*.DS_Store

--- a/src/jni/platforms/jvm_desktop.cpp
+++ b/src/jni/platforms/jvm_desktop.cpp
@@ -44,10 +44,15 @@ namespace jni {
 
         JavaVM* java_vm{nullptr};
         JNIEnv* jni_env{nullptr};
+
+#ifndef NO_USE_STDLIB
+        std::locale global;
+#endif
+
         jint result{JvmLoader::get_create_jvm_function()(&java_vm, reinterpret_cast<void**>(&jni_env), &args)};
 
 #ifndef NO_USE_STDLIB
-        std::locale::global(std::locale());
+        std::locale::global(global);
 #endif
 
         delete[] options;

--- a/src/jni/platforms/jvm_desktop.cpp
+++ b/src/jni/platforms/jvm_desktop.cpp
@@ -2,8 +2,12 @@
 
 #include <cassert>
 #include <modules/kotlin_jvm/src/logging.h>
+
+#ifndef NO_USE_STDLIB
+#include <locale>
+#endif
+
 #include "../jvm_loader.h"
-#include "../jvm.h"
 
 namespace jni {
     JavaVM* Jvm::vm = nullptr;
@@ -11,7 +15,7 @@ namespace jni {
     jint Jvm::version = 0;
 
     void Jvm::init(const InitArgs& initArgs) {
-        auto res = get_existing();
+        JavaVM* res{get_existing()};
         if (res == nullptr) {
             res = create(initArgs);
         }
@@ -38,12 +42,17 @@ namespace jni {
             args.options[i].optionString = (char*) initArgs.options[i].c_str();
         }
 
-        JavaVM* vm;
-        JNIEnv* env;
-        auto result = JvmLoader::get_create_jvm_function()(&vm, (void**) &env, (void*) &args);
+        JavaVM* java_vm{nullptr};
+        JNIEnv* jni_env{nullptr};
+        jint result{JvmLoader::get_create_jvm_function()(&java_vm, reinterpret_cast<void**>(&jni_env), &args)};
+
+#ifndef NO_USE_STDLIB
+        std::locale::global(std::locale());
+#endif
+
         delete[] options;
-        JVM_CRASH_COND_MSG(result != JNI_OK, "Failed to create a new vm!")
-        return vm;
+        JVM_CRASH_COND_MSG(result != JNI_OK, "Failed to create a new java_vm!")
+        return java_vm;
     }
 
     JavaVM* Jvm::get_existing() {

--- a/src/jni/platforms/jvm_desktop.cpp
+++ b/src/jni/platforms/jvm_desktop.cpp
@@ -56,7 +56,7 @@ namespace jni {
 #endif
 
         delete[] options;
-        JVM_CRASH_COND_MSG(result != JNI_OK, "Failed to create a new java_vm!")
+        JVM_CRASH_COND_MSG(result != JNI_OK, "Failed to create a new vm!")
         return java_vm;
     }
 

--- a/src/jni/platforms/jvm_desktop.cpp
+++ b/src/jni/platforms/jvm_desktop.cpp
@@ -51,6 +51,9 @@ namespace jni {
 
         jint result{JvmLoader::get_create_jvm_function()(&java_vm, reinterpret_cast<void**>(&jni_env), &args)};
 
+        // Set std::local::global to value it was before creating JVM.
+        // See https://github.com/utopia-rise/godot-kotlin-jvm/issues/166
+        // and https://github.com/utopia-rise/godot-kotlin-jvm/issues/170
 #ifndef NO_USE_STDLIB
         std::locale::global(global);
 #endif

